### ISSUE TITLE
StatusService.setOverwriting is not used

### ISSUE THREAD

**Issue by tliron** [restlet/restlet-framework-java#1073]
This value does not seem to be used anywhere.

It should either be implemented or deprecated.

**Comment by Tembrel** [comment]
The StatusFilter constructor calls `StatusService.isOverwriting` to provide an initial value for the service's `overwriting` property, which __is__ used:

https://github.com/restlet/restlet-framework-java/search?utf8=%E2%9C%93&q=isOverwriting&type=Code

**Comment by tliron** [comment]
Ah. Perhaps it's worth documenting this better in StatusService?

**Comment by Tembrel** [comment]
Indeed. I've changed the title to reflect that.

**Comment by thboileau** [comment]
Hi @tliron , @Tembrel , I've added a comment.
Does it suit you?

**Comment by tliron** [comment]
I think it would be a good idea to document this in StatusService, too, because the lack there is what brought me to open this issue.

Well, I've said this a few times before: Restlet JavaDocs are far too terse and could use a major beefing up, but that would be a whole project in itself.

**Comment by jlouvel** [comment]
I agree for the documentation front. The version 3 is the opportunity for a major overhaul. Here is what I have in mind:
  * revisit introduction (hello world, guided features tour)
  * rework user guide (move reference content to Javadocs, replace with tutorial/best practice/cookbook content)
  * improve javadocs (deeper content, consistent navigation bar with product logo, fix links)

This way, the Javadocs would become the sole **reference** documentation with much more the content currently available in the user guide (and more for missing parts). The user guide become more guidance material to help users being successful in their API projects based on RF (how to create, deploy, secure, manage, document, consume, etc.) with pointers to the Javadocs for reference materials.
